### ISSUE TITLE
Save to Geopackage fails with CPLE_AppDefinedError: failed: no such module: rtree

### ISSUE THREAD

**Issue by noenandre** [conda-forge/fiona-feedstock#53]
On a fresh Conda environment with Geopandas, Geopandas fails to save to a Geopackage file due to Fiona. 

```
CPLE_AppDefinedError                      Traceback (most recent call last)
<ipython-input-31-5263bde534b5> in <module>()
----> 1 geodataframe.to_file(driver='GPKG', filename='geopackage.gpkg', layer='geodata')

C:\Anaconda3\envs\geo\lib\site-packages\geopandas\geodataframe.py in to_file(self, filename, driver, schema, **kwargs)
    363         """
    364         from geopandas.io.file import to_file
--> 365         to_file(self, filename, driver, schema, **kwargs)
    366 
    367     def to_crs(self, crs=None, epsg=None, inplace=False):

C:\Anaconda3\envs\geo\lib\site-packages\geopandas\io\file.py in to_file(df, filename, driver, schema, **kwargs)
     64                         schema=schema, **kwargs) as colxn:
     65             for feature in df.iterfeatures():
---> 66                 colxn.write(feature)
     67 
     68 

C:\Anaconda3\envs\geo\lib\site-packages\fiona\collection.py in __exit__(self, type, value, traceback)
    419 
    420     def __exit__(self, type, value, traceback):
--> 421         self.close()
    422 
    423     def __del__(self):

C:\Anaconda3\envs\geo\lib\site-packages\fiona\collection.py in close(self)
    401         if self.session is not None:
    402             if self.mode in ('a', 'w'):
--> 403                 self.flush()
    404             log.debug("Flushed buffer")
    405             self.session.stop()

C:\Anaconda3\envs\geo\lib\site-packages\fiona\collection.py in flush(self)
    391         """Flush the buffer."""
    392         if self.session is not None:
--> 393             self.session.sync(self)
    394             new_len = self.session.get_length()
    395             self._len = new_len > self._len and new_len or self._len

fiona/ogrext.pyx in fiona.ogrext.WritingSession.sync()

fiona/_err.pyx in fiona._err.GDALErrCtxManager.__exit__()

CPLE_AppDefinedError: b'sqlite3_exec(CREATE VIRTUAL TABLE "rtree_geodata_geom" USING rtree(id, minx, maxx, miny, maxy)) failed: no such module: rtree'
```

The environment was created with "conda create -n geo -c conda-forge geopandas" on a windows 10 64-bit system, and the following versions of packages was installed:
```
# packages in environment at C:\Anaconda3\envs\geo:
#
fiona                     1.7.9                    py36_1    conda-forge
geopandas                 0.3.0                    py36_0    conda-forge
rtree                     0.8.3                    py36_0    conda-forge

```

**Comment by ocefpaf** [comment]
Can you send the full package list? One missing key there is the `gdal` BTW.

Also, can you put together a [sscce](http://sscce.org)?

**Comment by sgillies** [comment]
This has nothing to do with fiona or the rtree python package. See section 2 of https://www.sqlite.org/rtree.html.

**Comment by ocefpaf** [comment]
Thanks for the info @sgillies.

@noenandre in light of that the only thing we can do is to send a PR to the `sqlite` feedstock to fix that. Do you want to give it a try?

**Comment by ocefpaf** [comment]
Closing this as the fix belongs somewhere else.

**Comment by double-u-a** [comment]
Still appear to have this error when saving to geopackage with Fiona 1.7.13 on Windows 10, does this version use the updated sqlite with rtree module? Or is my environment incorrectly setup?

`conda list`
```
altair                    2.2.2                    py36_1    conda-forge
appdirs                   1.4.3            py36h28b3542_0
asn1crypto                0.24.0                   py36_0
attrs                     18.2.0           py36h28b3542_0
automat                   0.7.0                    py36_0
backcall                  0.1.0                    py36_0
blas                      1.0                         mkl
bleach                    2.1.4                    py36_0
boost                     1.66.0              py36_vc14_1  [vc14]  conda-forge
boost-cpp                 1.66.0                   vc14_1  [vc14]  conda-forge
branca                    0.3.0                      py_0    conda-forge
ca-certificates           2018.8.24            ha4d7672_0    conda-forge
certifi                   2018.8.24                py36_0    conda-forge
cffi                      1.11.5           py36h945400d_0
chardet                   3.0.4            py36h420ce6e_1
click                     6.7                        py_1    conda-forge
click-plugins             1.0.3                      py_1    conda-forge
cligj                     0.4.0                      py_1    conda-forge
colorama                  0.3.9            py36h029ae33_0
console_shortcut          0.1.1                h6bb2dd7_3
constantly                15.1.0           py36h28b3542_0
cryptography              2.2.2            py36hfa6e2cd_0
curl                      7.60.0                   vc14_0  [vc14]  conda-forge
cycler                    0.10.0                     py_1    conda-forge
decorator                 4.3.0                    py36_0
descartes                 1.1.0                      py_2    conda-forge
entrypoints               0.2.3                    py36_2
et_xmlfile                1.0.1            py36h3d2d736_0
expat                     2.2.5                    vc14_0  [vc14]  conda-forge
fiona                     1.7.13           py36h01e27f7_3    conda-forge
folium                    0.6.0                      py_0    conda-forge
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
freexl                    1.0.2                    vc14_2  [vc14]  conda-forge
gdal                      2.2.4            py36h2fc6367_1    conda-forge
geopandas                 0.4.0                      py_1    conda-forge
geos                      3.6.2                he025d50_3    conda-forge
geotiff                   1.4.2                    vc14_1  [vc14]  conda-forge
graphviz                  2.38                 hfd603c8_2
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
html5lib                  1.0.1                    py36_0
hyperlink                 18.0.0                   py36_0
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
idna                      2.6              py36h148d497_1
incremental               17.5.0                   py36_0
intel-openmp              2018.0.3                      0
ipykernel                 4.9.0                    py36_0
ipython                   6.5.0                    py36_0
ipython_genutils          0.2.0            py36h3c5d0ee_0
jdcal                     1.4                      py36_0
jedi                      0.12.1                   py36_0
jinja2                    2.10                     py36_0
jpeg                      9b                       vc14_2  [vc14]  conda-forge
jsonschema                2.6.0            py36h7636477_0
jupyter_client            5.2.3                    py36_0
jupyter_core              4.4.0                    py36_0
jupyterlab                0.34.9                   py36_0
jupyterlab_launcher       0.13.1                   py36_0
kealib                    1.4.7                    vc14_4  [vc14]  conda-forge
kiwisolver                1.0.1            py36he980bc4_2    conda-forge
krb5                      1.14.6                   vc14_0  [vc14]  conda-forge
libgdal                   2.2.4                    vc14_5  [vc14]  conda-forge
libiconv                  1.14                     vc14_4  [vc14]  conda-forge
libnetcdf                 4.6.1                    vc14_2  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libpq                     9.6.3                    vc14_0  [vc14]  conda-forge
libsodium                 1.0.16                   vc14_0  [vc14]  conda-forge
libspatialindex           1.8.5                    vc14_1  [vc14]  conda-forge
libspatialite             4.3.0a                  vc14_19  [vc14]  conda-forge
libssh2                   1.8.0                hc4dcbb0_2    conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
libxml2                   2.9.5                    vc14_1  [vc14]  conda-forge
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
markupsafe                1.0              py36hfa6e2cd_1
matplotlib                2.2.2                    py36_1    conda-forge
menuinst                  1.4.14           py36hfa6e2cd_0
mistune                   0.8.3            py36hfa6e2cd_1
mkl                       2018.0.3                      1
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.1                    py36_0    conda-forge
msys2-conda-epoch         20160418                      1
munch                     2.3.2                      py_0    conda-forge
nbconvert                 5.3.1                    py36_0
nbformat                  4.4.0            py36h3a5bc1b_0
notebook                  5.6.0                    py36_0
numpy                     1.15.1           py36ha559c80_0
numpy-base                1.15.1           py36h8128ebf_0
openjpeg                  2.3.0                    vc14_2  [vc14]  conda-forge
openpyxl                  2.5.6                    py36_0
openssl                   1.0.2p               hfa6e2cd_0    conda-forge
pandas                    0.23.4           py36h830ac7b_0
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py36_1
parso                     0.3.1                    py36_0
pickleshare               0.7.4            py36h9de030f_0
pip                       10.0.1                   py36_0
proj4                     4.9.3                    vc14_5  [vc14]  conda-forge
prometheus_client         0.3.1            py36h28b3542_0
prompt_toolkit            1.0.15           py36h60b8f86_0
psycopg2                  2.7.5            py36h74b6da3_1    conda-forge
pyasn1                    0.4.4            py36h28b3542_0
pyasn1-modules            0.2.2                    py36_0
pycosat                   0.6.3            py36h413d8a4_0
pycparser                 2.18             py36hd053e01_1
pygments                  2.2.0            py36hb010967_0
pyopenssl                 18.0.0                   py36_0
pyparsing                 2.2.0                      py_1    conda-forge
pyproj                    1.9.5.1          py36haf3c679_3    conda-forge
pyqt                      5.6.0            py36h764d66f_7    conda-forge
pysal                     1.14.4.post2             py36_0    conda-forge
pysocks                   1.6.8                    py36_0
python                    3.6.5                h0c2934d_0
python-dateutil           2.7.3                      py_0    conda-forge
python-graphviz           0.8.4                    py36_1
pytz                      2018.5                     py_0    conda-forge
pywin32                   223              py36hfa6e2cd_1
pywinpty                  0.5.4                    py36_0
pyzmq                     17.1.2           py36hfa6e2cd_0
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
requests                  2.18.4           py36h4371aae_1
rtree                     0.8.3                    py36_0    conda-forge
ruamel_yaml               0.15.37          py36hfa6e2cd_2
scipy                     1.1.0            py36h4f6bf74_1
send2trash                1.5.0                    py36_0
service_identity          17.0.0           py36h28b3542_0
setuptools                39.2.0                   py36_0
shapely                   1.6.4            py36hc90234e_1    conda-forge
simplegeneric             0.8.1                    py36_2
sip                       4.18.1           py36h6538335_0    conda-forge
six                       1.11.0           py36h4db2310_1
sqlalchemy                1.2.11           py36hfa6e2cd_0    conda-forge
sqlite                    3.20.1                   vc14_2  [vc14]  conda-forge
terminado                 0.8.1                    py36_1
testpath                  0.3.1            py36h2698cfe_0
toolz                     0.9.0                      py_0    conda-forge
tornado                   5.1              py36hfa6e2cd_1    conda-forge
traitlets                 4.3.2            py36h096827d_0
twisted                   18.7.0           py36hfa6e2cd_1
urllib3                   1.22             py36h276f60a_0
vc                        14                   h0510ff6_3
vincent                   0.4.4                      py_1    conda-forge
vs2015_runtime            14.0.25123                    3
wcwidth                   0.1.7            py36h3d5aa90_0
webencodings              0.5.1                    py36_1
wheel                     0.31.1                   py36_0
win_inet_pton             1.0.1            py36he67d7fd_1
wincertstore              0.2              py36h7fe50ca_0
winpty                    0.4.3                         4
xerces-c                  3.2.0                    vc14_0  [vc14]  conda-forge
xlrd                      1.1.0                    py36_1
xz                        5.2.4             h2fa13f4_1001    conda-forge
yaml                      0.1.7                    vc14_0  [vc14]  conda-forge
zeromq                    4.2.5                    vc14_2  [vc14]  conda-forge
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
zope                      1.0                      py36_1
zope.interface            4.5.0            py36hfa6e2cd_0
```

**Comment by moekloar** [comment]
I would like to note, that when i try to install fiona from conda-forge channel i still get this error on windows 10... For me it can be solved if i use the fiona (v1.8.2) and GDAL (2.3.2) from [Christof Gohlke's unofficial python extensions] (https://www.lfd.uci.edu/~gohlke/pythonlibs/) via pip wheel, though i find this behavior really strange and not very intuitive. I have to say, that i do not know what he adds to his wheels to make it work, but maybe it helps people with this problem.

@sgillies I would like to know how this issue can be resolved if its not a fiona or rtree issue? I mean, when i just install fiona with windows specific binaries and do not update sqlite3 nor rtree and it works after that where else is the issue? Is there a way to enable rtree in condas sqlite without recompiling?

`conda list` of my now working environment

```
# Name                    Version                   Build  Channel
attrs                     18.2.0                     py_0    conda-forge
attrs                     18.2.0                    <pip>
backcall                  0.1.0                      py_0    conda-forge
blas                      1.0                         mkl
bleach                    3.0.2                      py_0    conda-forge
boost                     1.66.0              py36_vc14_1  [vc14]  conda-forge
boost-cpp                 1.66.0                   vc14_1  [vc14]  conda-forge
bzip2                     1.0.6                    vc14_1  [vc14]  conda-forge
ca-certificates           2018.10.15           ha4d7672_0    conda-forge
certifi                   2018.10.15            py36_1000    conda-forge
click                     7.0                        py_0    conda-forge
Click                     7.0                       <pip>
click-plugins             1.0.4                      py_0    conda-forge
cligj                     0.5.0                     <pip>
cligj                     0.5.0                      py_0    conda-forge
colorama                  0.4.0                      py_0    conda-forge
curl                      7.60.0                   vc14_0  [vc14]  conda-forge
decorator                 4.3.0                      py_0    conda-forge
entrypoints               0.2.3                 py36_1002    conda-forge
expat                     2.2.5                    vc14_0  [vc14]  conda-forge
fiona                     1.7.13           py36h2fc6367_4    conda-forge
Fiona                     1.8.2                     <pip>
freexl                    1.0.2                    vc14_2  [vc14]  conda-forge
gdal                      2.2.4            py36h2fc6367_1    conda-forge
GDAL                      2.3.2                     <pip>
geos                      3.6.2                he025d50_3    conda-forge
geotiff                   1.4.2                    vc14_1  [vc14]  conda-forge
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
intel-openmp              2019.1                      144
ipykernel                 5.1.0           py36h39e3cac_1001    conda-forge
ipython                   7.1.1           py36h39e3cac_1000    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.2                      py_0    conda-forge
jedi                      0.13.1                py36_1000    conda-forge
jinja2                    2.10                       py_1    conda-forge
jpeg                      9b                       vc14_2  [vc14]  conda-forge
jsonschema                3.0.0a3               py36_1000    conda-forge
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                      py_1    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
kealib                    1.4.7                    vc14_4  [vc14]  conda-forge
krb5                      1.14.6                   vc14_0  [vc14]  conda-forge
libboost                  1.67.0               hd9e427e_4
libgdal                   2.2.4                    vc14_5  [vc14]  conda-forge
libiconv                  1.14                     vc14_4  [vc14]  conda-forge
libkml                    1.3.0                he5f2a48_4
libnetcdf                 4.6.1                    vc14_2  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libpq                     9.6.3                    vc14_0  [vc14]  conda-forge
libsodium                 1.0.16                   vc14_0  [vc14]  conda-forge
libspatialindex           1.8.5                    vc14_1  [vc14]  conda-forge
libspatialite             4.3.0a                  vc14_19  [vc14]  conda-forge
libssh2                   1.8.0             hc4dcbb0_1003    conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
libxml2                   2.9.5                    vc14_1  [vc14]  conda-forge
m2w64-expat               2.1.1                         2
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gettext             0.19.7                        2
m2w64-gmp                 6.1.0                         2
m2w64-libiconv            1.14                          6
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
m2w64-xz                  5.2.2                         2
markupsafe                1.1.0           py36hfa6e2cd_1000    conda-forge
mistune                   0.8.4           py36hfa6e2cd_1000    conda-forge
mkl                       2018.0.3                      1
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
msys2-conda-epoch         20160418                      1
munch                     2.3.2                      py_0    conda-forge
munch                     2.3.2                     <pip>
nbconvert                 5.3.1                      py_1    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
notebook                  5.7.2                 py36_1000    conda-forge
numpy                     1.15.4           py36ha559c80_0
numpy-base                1.15.4           py36h8128ebf_0
openjpeg                  2.3.0                    vc14_2  [vc14]  conda-forge
openssl                   1.0.2p            hfa6e2cd_1001    conda-forge
pandoc                    2.4                           0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.3.1                      py_0    conda-forge
pcre                      8.41                     vc14_1  [vc14]  conda-forge
pickleshare               0.7.5                 py36_1000    conda-forge
pip                       18.1                  py36_1000    conda-forge
postgresql                9.6.3                    vc14_4  [vc14]  conda-forge
proj4                     4.9.3                    vc14_5  [vc14]  conda-forge
prometheus_client         0.4.2                      py_0    conda-forge
prompt_toolkit            2.0.7                      py_0    conda-forge
pygments                  2.2.0                      py_1    conda-forge
pyproj                    1.9.5.1         py36haf3c679_1006    conda-forge
pyqt                      5.6.0           py36h764d66f_1007    conda-forge
pyrsistent                0.14.7          py36hfa6e2cd_1000    conda-forge
python                    3.6.6                he025d50_0    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pywinpty                  0.5.4                 py36_1002    conda-forge
pyzmq                     17.1.2          py36hf576995_1001    conda-forge
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
qtconsole                 4.4.3                      py_0    conda-forge
rtree                     0.8.3                 py36_1000    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                40.6.2                   py36_0    conda-forge
shapely                   1.6.4           py36hc90234e_1000    conda-forge
sip                       4.18.1           py36h6538335_0    conda-forge
six                       1.11.0                py36_1001    conda-forge
sqlite                    3.20.1                   vc14_2  [vc14]  conda-forge
terminado                 0.8.1                 py36_1001    conda-forge
testpath                  0.4.2                 py36_1000    conda-forge
tornado                   5.1.1           py36hfa6e2cd_1000    conda-forge
traitlets                 4.3.2                 py36_1000    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.0.25420                    0    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.32.3                   py36_0    conda-forge
widgetsnbextension        3.4.2                 py36_1000    conda-forge
wincertstore              0.2                   py36_1002    conda-forge
winpty                    0.4.3                         4    conda-forge
xerces-c                  3.2.0                    vc14_0  [vc14]  conda-forge
xz                        5.2.4             h2fa13f4_1001    conda-forge
zeromq                    4.2.5                    vc14_2  [vc14]  conda-forge
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```
